### DISCUSSION TITLE
Setting temperature when calling Deepgram's Whisper endpoint.

### DISCUSSION THREAD

**Discussion by troublesprouter** [deepgram/community#198] [General help] [answered]
I don't see an option to set the temperature, which is key and necessary for my application. Is it possible to set the temperature when calling Deepgram's Whisper endpoint?

**Comment by jjmaldonis** [comment, marked as answer]
Hi @troublesprouter, setting the temperature for whisper is not currently possible. What is your use case? I'd like to pass your request along to the team and more info would be helpful.

**Comment by troublesprouter** [comment]
Thank you. Without going into too much detail at this stage, it's a consumer application that on testing we've found the temperature to be critical for accuracy. For now I'll keep looking for Whisper APIs or possibly an "in-house" deployment but do you know what temperature you currently use?

I haven't looked into it yet partly because I'd rather use it via APIs but I assume the code that was open sourced allows setting temperature just like the OpenAI API?

I imagine adding temperature should be straightforward in your setup, along the lines of

```
  response = await asyncio.create_task(
    deepgram.transcription.prerecorded(
      source,
      {
        'smart_format': True,
        'model': 'whisper-large',
        'temperature': 0.5,
      }
    )
  )
```

**Comment by jjmaldonis** [comment]
This info is great - I'll pass it along to our team to keep in mind as we develop our next-gen models. If you'd like to talk 1-1 and see if there is some other way we can meet your accuracy needs, feel free to send me an email at jason.maldonis at deepgram dot com. I'm an Applied Engineer at Deepgram and we are always interested in hearing how our users are leveraging our API to do cool things like what you alluded to.

As you mentioned, temperature can play a key role in accuracy. Our goal at Deepgram is to optimize these kinds of hyperparameters for you so you don't need to go through the trouble of doing it yourself, and in the case of speech-to-text it's possible that a fixed value for the temperature may be appropriate for almost everyone. If you've found something different for your use case, we'd be very interested in hearing more!

In the mean time, I'll try to track down the temperature value we are currently using.

**Comment by troublesprouter** [comment]
Setting automatic defaults makes sense for people who don't want to tinker, but in production at least in my case temperature setting is key for accuracy to the point that the inability to set it is the single thing that prevents me from using the service.

If you add the optional temperature parameter to the API please let me know and I'll gladly use it. Unfortunately I've tested and the results of default temperature are unusable for my use case.

**Comment by misters2008** [comment]
Temperature, temperature incerement on fallback, batch size and no_gram words would be useful for those, who really want quality.
i ve found a good mix of whisper settings using features above, but i only lack a good enough dizrization, which Whisper does not have.
However Deepgram's diarization is good enough.
But the transcription is not. in my cases . With Whisper-large using Deepgram it gives a very close result to what i want with diariztion.
But the abscence of possibility to alter the setting of the advanced features in the first line of my reponse here is really stopping me.

i want to transcribe close to 3500 hours, but i cannot procceed at the moment.

In, addition, dont really get it, why not to let users alter the advanced features like those. (Who needs the, will use them, who doesnt need, simply will use the default).
